ambari-server/src/main/java/org/apache/ambari/server/controller/internal/BlueprintConfigurationProcessor.java

```diff
@@ -406,21 +412,70 @@ private boolean containsHostFromHostGroups(String configType, String propertyNam
-        Collection<String> nnHosts = clusterTopology.getHostAssignmentsForComponent("NAMENODE");
-        if (nnHosts.size() < 2) {
-          throw new ConfigurationTopologyException("NAMENODE HA requires at least 2 hosts running NAMENODE but there are: " +
-              nnHosts.size() + " Hosts: " + nnHosts);
+      // parse out the nameservices value
+      String[] parsedNameServices = parseNameServices(hdfsSiteConfig);
+
+      // if a single nameservice is configured (default HDFS HA deployment)
+      if (parsedNameServices.length == 1) {
+        LOG.info("Processing a single HDFS NameService, which indicates a default HDFS NameNode HA deployment");
+        // if the active/stanbdy namenodes are not specified, assign them automatically
+        if (! isNameNodeHAInitialActiveNodeSet(clusterProps) && ! isNameNodeHAInitialStandbyNodeSet(clusterProps)) {
+          Collection<String> nnHosts = clusterTopology.getHostAssignmentsForComponent("NAMENODE");
+          if (nnHosts.size() < 2) {
+            throw new ConfigurationTopologyException("NAMENODE HA requires at least 2 hosts running NAMENODE but there are: " +
+                nnHosts.size() + " Hosts: " + nnHosts);
+          }
+
+          // set the properties that configure which namenode is active,
+          // and which is a standby node in this HA deployment
+          Iterator<String> nnHostIterator = nnHosts.iterator();
+          clusterConfig.setProperty("hadoop-env", "dfs_ha_initial_namenode_active", nnHostIterator.next());
+          clusterConfig.setProperty("hadoop-env", "dfs_ha_initial_namenode_standby", nnHostIterator.next());
+
+          configTypesUpdated.add("hadoop-env");
         }
+      } else {
+        if (!isPropertySet(clusterProps, "hadoop-env", HDFS_ACTIVE_NAMENODE_SET_PROPERTY_NAME) && !isPropertySet(clusterProps, "hadoop-env", HDFS_STANDBY_NAMENODE_SET_PROPERTY_NAME)) {
+          // multiple nameservices indicates an HDFS NameNode Federation install
+          // process each nameservice to determine the active/standby nodes
+          LOG.info("Processing multiple HDFS NameService instances, which indicates a NameNode Federation deployment");
+          if (parsedNameServices.length > 1) {
+            Set<String> activeNameNodeHostnames = new HashSet<String>();
+            Set<String> standbyNameNodeHostnames = new HashSet<String>();
+
+            for (String nameService : parsedNameServices) {
+              List<String> hostNames = new LinkedList<String>();
+              String[] nameNodes = parseNameNodes(nameService, hdfsSiteConfig);
+              for (String nameNode : nameNodes) {
+                // use the HA rpc-address property to obtain the NameNode hostnames
+                String propertyName = "dfs.namenode.rpc-address." + nameService + "." + nameNode;
+                String propertyValue = hdfsSiteConfig.get(propertyName);
+                String hostName = propertyValue.split(":")[0];
+                hostNames.add(hostName);
```

Comment:
Good catch.  I've added a check for null, and a new exception will be thrown when this situation is detected, since the absence of this property indicates an inconsistent configuration in hdfs-site.

Reply:
I'm also adding a unit test to verify this in the updated patch as well.  thanks!